Applications/Utils/ModelPreparation/PartitionMesh/NodeWiseMeshPartitioner.cpp

```diff
@@ -0,0 +1,564 @@
+/*!
+  \file NodeWiseMeshPartitioner.cpp
+  \date   2016.05
+
+  \brief  Define the members of class NodeWiseMeshPartitioner
+
+  \copyright
+  Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+             Distributed under a Modified BSD License.
+               See accompanying file LICENSE.txt or
+               http://www.opengeosys.org/project/license
+
+*/
+
+#include "NodeWiseMeshPartitioner.h"
+
+#include <limits>
+#include <iomanip>
+#include <cstdio>  // for binary output
+
+#include <logog/include/logog.hpp>
+
+#include "BaseLib/Error.h"
+
+#include "MeshLib/IO/VtkIO/VtuInterface.h"
+
+#include "MeshLib/Node.h"
+#include "MeshLib/Elements/Element.h"
+
+namespace ApplicationUtils
+{
+struct NodeStruct
+{
+    NodeWiseMeshPartitioner::IntegerType id;
+    double x;
+    double y;
+    double z;
+};
+
+void NodeWiseMeshPartitioner::readMetisData(const std::string& file_name_base)
+{
+    const std::string npartitions_str = std::to_string(_npartitions);
+
+    // Read partitioned mesh data from METIS
+    const std::string fname_parts = file_name_base + ".mesh.npart." + npartitions_str;
+
+    std::ifstream npart_in(fname_parts);
+    if (!npart_in.is_open())
+    {
+        OGS_FATAL(
+            "Error: cannot open file %s. It may not exist! \
+                   \n Run mpmetis beforehand or use option -m",
+            fname_parts.data());
+    }
+
+    const std::size_t nnodes = _mesh->getNumberOfNodes();
+
+    std::size_t counter = 0;
+    while (!npart_in.eof())
+    {
+        npart_in >> _nodes_partition_ids[counter++] >> std::ws;
+        if (counter == nnodes)
+          break;
+    }
+
+    if (npart_in.bad())
+    {
+        OGS_FATAL(
+            "Error while reading file %s.", fname_parts.data());
+    }
+
+    npart_in.close();
+
+    if( counter != nnodes)
+    {
+        OGS_FATAL(
+            "Error: data in %s are less than expected.", fname_parts.data());
+    }
+
+    // TEST  std::remove(fname_parts.c_str());
```

Comment:
I just wondered before that whether the files generated by metis should be kept.
Finally, I decide to remove these files at the end of this function as
```
    std::remove(fname_parts.c_str());
    const std::string fname_eparts = file_name_base + ".mesh.epart."
                                     + npartitions_str;
    std::remove(fname_eparts.c_str());
```